src/js/client/pages/Cookbook.js

```diff
@@ -18,166 +18,214 @@ export default class Cookbook extends React.Component {
       recipeDetail: {
         isActive: false,
         recipe: {}
-      },
+      }
     };
   }
 
   componentWillMount() {
     let { appState } = this.props;
-    if(this.props.currentUser) {
-      httpClient.userRecipes(this.props.currentUser._id).then((response) => {
+    // if (this.props.currentUser) {
+    //   httpClient.userRecipes(this.props.currentUser._id).then(response => {
+    //     this.setState({
+    //       userRecipes: response.data
+    //     });
+    //   });
+  }
+
+  componentDidMount() {
+    if (this.props.currentUser) {
+      httpClient.userRecipes(this.props.currentUser._id).then(response => {
         this.setState({
           userRecipes: response.data
         });
+        console.log(this.state.userRecipes);
       });
     }
   }
 
-  componentDidMount (){
-
-  }
+  // updateRecipes(newRecipes) {
+  //   this.setState({ userRecipes: newRecipes });
+  // }
 
-  updateRecipes(newRecipes) {
-    this.setState({recipeForm: newRecipes});
+  addNewRecipe(recipe) {
+    this.setState({
+      userRecipes: [...this.state.userRecipes, recipe]
+    });
+    console.log(this.state.userRecipes);
   }
 
-  render () {
+  render() {
     let { appState, updateAppState } = this.props;
     let { recipe, isActive, idx } = this.state.recipeDetail;
-    if(appState.currentUser){
+    if (appState.currentUser) {
       return (
         <section class="c-cookbook">
           <h1 class="c-cookbook__heading">Cookbook</h1>
           <div class="c-cookbook__page container">
-            {this.renderUserRecipes()}
+            {this.renderUserRecipes(this.state.userRecipes)}
             <RecipeDetail
               recipe={recipe}
               isActive={isActive}
               renderIngredientList={this.renderIngredientList}
               renderInstructionsList={this.renderInstructionsList}
-              deleteRecipe={this.deleteRecipe.bind(this, recipe, idx)}
-              handleToggleRecipeDetail={this.handleToggleRecipeDetail.bind(this, recipe, idx) } />
+              userRecipes={this.state.userRecipes}
+              deleteRecipe={this.deleteRecipe.bind(this)}
+              handleToggleRecipeDetail={this.handleToggleRecipeDetail.bind(
+                this,
+                recipe,
+                idx
```

Comment:
lol this breaking each parameter to each line is a little extra but i love all this 👍